### DISCUSSION TITLE
Why not cache .so and .a files?

### DISCUSSION THREAD

**Discussion by vaneyli** [ccache/ccache#241] [Ideas]
HI guys:
Two questions: 
1:some .so cost a long time to link and some huge projects have lots of .so.

I think it will be a good feature to cache .so and .a.
2: Will these be a distributed ccache?  The NFS will be a good way but the manual shows that ccache is not fully tested in NFS.

Thanks

**Comment by jrosdahl** [comment]
> 1:some .so cost a long time to link and some huge projects have lots of .so.

It's not trivial to make ccache support that as well. And I'm not convinced that it's worth the effort even if it were relatively easy. See also https://www.mail-archive.com/ccache@lists.samba.org/msg00870.html for a previous discussion of this topic.

> 2: Will these be a distributed ccache? The NFS will be a good way but the manual shows that ccache is not fully tested in NFS.

I think that sharing a cache NFS works well if you have configured NFS correctly, but I don't use it personally. You can certainly try it out if you want.

There is ongoing work on sharing a cache via memcached, but it has not reached any released ccache version yet.

**Comment by afbjorklund** [comment]
I think it ("ldcache") would be an interesting feature to have, but it might even be a separate project...

Having ccache on NFS does work, but it is a bit under-documented. There are some tricks to config.
